dotCMS/src/main/java/com/dotmarketing/portlets/rules/util/RulePermissionableUtil.java

```diff
@@ -37,9 +37,15 @@ public static Permissionable findParentPermissionable(String parent) throws DotD
                 if(iden.getAssetType().equals("folder")){
                     permissionableParent = APILocator.getFolderAPI().find(parent,systemUser,false);
                 }else{
-					Contentlet contentlet = APILocator.getContentletAPI().findContentletByIdentifier(parent, false,
-							APILocator.getLanguageAPI().getDefaultLanguage().getId(), systemUser, false);
-					if (contentlet.isHost()) {
+                    Contentlet contentlet;
+                    try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)